site/public/js/user-profile.js

```diff
@@ -235,11 +235,15 @@ function updateUserPreferredNames() {
                     displaySuccessMessage(data.message);
                     // update the preferred names
                     const icon = '<i class="fas fa-pencil-alt"></i>';
-                    $('#givenname-row .icon').html(`${icon} ${data.given_name}`);
-                    $('#familyname-row .icon').html(`${icon} ${data.family_name}`);
+                    $('#givenname-row .icon').html(`${icon} ${data.displayed_given_name}`);
+                    $('#familyname-row .icon').html(`${icon} ${data.displayed_family_name}`);
                     // update the data attributes
-                    given_name_field.data('current-name', data.given_name);
-                    family_name_field.data('current-name', data.family_name);
+                    given_name_field.data('current-name', data.preferred_given_name);
+                    family_name_field.data('current-name', data.preferred_family_name);
+                    // Update abbreviated name
+                    $('#user-last-initial-format-preview').data('options', data.abbreviation_options);
+                    $('#user-last-initial-format-preview').text(data.current_abbreviation);
+                    $('#last-initial-format-row .icon').html(`${icon} ${data.current_abbreviation}`);
```

Comment:
What's the purpose of `.html()` here?  Directly injecting user-controllable text strings into the DOM opens up the possibility of XSS attacks.

Reply:
What about leaving the icon in the DOM and just hiding/showing it as needed?

Reply:
The icon always stays, the reason it's getting replaced every time seems to just be avoiding wrapping the text in an element, before Friday I'll update the code to start with the text in a span and only edit that span